chains/evm/calls/events/listener.go

```diff
@@ -34,32 +32,29 @@ func NewListener(client ChainClient) *Listener {
 }
 
 func (l *Listener) FetchDeposits(ctx context.Context, contractAddress common.Address, startBlock *big.Int, endBlock *big.Int) ([]*Deposit, error) {
-	ctxWithSpan, span := otel.Tracer("relayer-core").Start(ctx, "relayer.core.Listener.FetchDeposits")
+	ctx, span, logger := observability.CreateSpanAndLoggerFromContext(ctx, "relayer-core", "relayer.core.Listener.FetchDeposits", attribute.String("startBlock", startBlock.String()), attribute.String("endBlock", endBlock.String()))
 	defer span.End()
-	span.SetAttributes(attribute.String("startBlock", startBlock.String()), attribute.String("endBlock", endBlock.String()))
-	logger := log.With().Str("dd.trace_id", span.SpanContext().TraceID().String()).Logger()
 
-	logs, err := l.client.FetchEventLogs(ctxWithSpan, contractAddress, string(DepositSig), startBlock, endBlock)
+	logs, err := l.client.FetchEventLogs(ctx, contractAddress, string(DepositSig), startBlock, endBlock)
 	if err != nil {
-		span.SetStatus(codes.Error, err.Error())
-		return nil, err
+		return nil, observability.LogAndRecordErrorWithStatus(&logger, span, err, "failed FetchEventLogs")
```

Comment:
Most of the retuns are overkill.
It is really not intuitive to log and record every error for every return path when you log it as well when it is propagated.

We didn't log them before and setting span status is overkill as we set span status in the event handler anyway.
I would personally remove the the log and record functions.
If we need both I would prefer to have it explicitly.

Reply:
It is not Logging every error, in cases when an error is going to bubble up higher you can just pass nil to the logger, it is described in func readme.
This one really goes up the stack so I will replace logger with nil

Reply:
I am more talking about the logic specifically, not necessarily what is logged and traced.
We don't need to trace every error as well as we also have that in the parent span.

Reply:
I do not agree, it is pretty useful to mark errored span, since they are not linear and one parent span can hame multiple children, UI nicely displays errored spans with colours which dramatically improves debugging UX

Reply:
I agree with you generally, but you have the weigh the options of future maintability and code cleanliness with the information gained.

You see the error anyway where it happened because it is always from the last span which is nicely visualized.

Ideally we would trace every action that the system makes if there is no penalty, but there obviously is.

Reply:
Yeah, but it submits traces in batches within a designated time period, so adding more traces is not going to dramatically influence performance.

Reply:
I was more talking about the cognitive load needed for someone to understand why do we have a helper function for returning errors.  
It is a future nightmare to maintain the code and it is not very clear why do we have it..

Reply:
the function itself is pretty clear I do not see any reason that someone would not understand what is it doing.